aws/PAS-AIO-template.yaml

```diff
@@ -0,0 +1,952 @@
+                Resource: '*'
+        - PolicyName: VaultBootstrapIAMPolicy
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Action:
+                  - iam:DeleteRolePolicy
+                  - iam:PutRolePolicy
+                Resource: '*'
+    Condition: EULACondition
+  InstancesSSMPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: InstancesSsmAccess
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - ssm:GetParameter
+            Resource:
+              - !Sub 'arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter/*'
+      Roles:
+        - !Ref 'ComponentInstanceRole'
+        - !Ref 'VaultInstancesRole'
+    Condition: EULACondition
+  VaultInstancesKMSPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: VaultInstancesKMSAccess
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - kms:Encrypt
+              - kms:Decrypt
+            Resource: '*'
+      Roles:
+        - !Ref 'VaultInstancesRole'
+    Condition: EULACondition
+  VaultInstancesS3VaultFilesBucketPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: VaultFilesBucketAccess
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - s3:GetObject
+              - s3:GetObjectVersion
+            Resource: !Sub 'arn:${AWS::Partition}:s3:::${VaultFilesBucket}/*'
+      Roles:
+        - !Ref 'VaultInstancesRole'
+    Condition: EULACondition
+  VaultBootstrapKMSPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: VaultBootstrapKMSAccess
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - kms:CreateKey
+              - kms:GenerateRandom
+            Resource: '*'
+      Roles:
+        - !Ref 'VaultInstancesRole'
+  ComponentsMachine:
+    DependsOn: 
+      - VaultMachine
+    Type: AWS::EC2::Instance
+    Properties:
+      Tags:
+        - Key: Name
+          Value: !Ref 'ComponentsInstanceName'
+      SecurityGroupIds: !Ref 'ComponentsInstanceSecurityGroups'
+      SubnetId: !Ref 'ComponentsInstanceSubnetId'
+      ImageId: !FindInMap
+        - RegionMap
+        - !Ref 'AWS::Region'
+        - Components
+      InstanceType: !Ref 'ComponentsInstanceType'
+      UserData: !Base64
+        Fn::Sub: |-
+          <script>
+          cfn-init.exe -v -s ${AWS::StackId} -r ComponentsMachine --configsets ascending --region ${AWS::Region}
+          </script>
+      KeyName: !Ref 'KeyName'
+      IamInstanceProfile: !Ref 'ComponentInstanceProfile'
+    Metadata:
+      AWS::CloudFormation::Init:
+        configSets:
+          ascending:
+            - configSSMAndHostname
+            - configPVWA
+            - configCPM
+            - configPSM
+            - configSignal
+        configSSMAndHostname:
```

Comment:
change the name here and in every corresponding place.
we don't change the hostname in AIO
add restart to ssm like in other templates